warp10/src/main/java/io/warp10/script/filter/FilterAny.java

```diff
@@ -0,0 +1,165 @@
+      this.complementSet = complement;
+    }
+
+    public Builder(String name, Comparator comparator) {
+      super(name);
+      this.comparator = comparator;
+      this.complementSet = false;
+    }
+
+    @Override
+    public Object apply(WarpScriptStack stack) throws WarpScriptException {
+      Object threshold = stack.pop();
+      stack.push(new FilterAny(getName(), threshold, this.comparator, this.complementSet));
+      return stack;
+    }
+  }
+
+  public FilterAny(String name, Object threshold, Comparator comparator, boolean complementSet) throws WarpScriptException {
+    super(name);
+    this.comparator = comparator;
+    this.complementSet = complementSet;
+
+    boolean allowBooleanThreshold = comparator == Comparator.EQ || comparator == Comparator.NE;
+
+    if (threshold instanceof Long) {
+      this.type = TYPE.LONG;
+      this.threshold = threshold;
+    } else if (threshold instanceof Double) {
+      this.type = TYPE.DOUBLE;
+      this.threshold = threshold;
+    } else if (threshold instanceof String) {
+      this.type = TYPE.STRING;
+      this.threshold = threshold;
+    } else if (allowBooleanThreshold && threshold instanceof Boolean) {
+      this.type = TYPE.BOOLEAN;
+      this.threshold = threshold;
+    } else {
+      throw new WarpScriptException(getName() + " threshold type is invalid.");
+    }
+  }
+
+  private boolean verify(int i) throws WarpScriptException {
+    switch (this.comparator) {
+      case EQ:
+        return i == 0;
+      case GE:
+        return i <= 0;
+      case GT:
+        return i < 0;
+      case LE:
+        return i >= 0;
+      case LT:
+        return i > 0;
+      case NE:
+        return i != 0;
+      default:
+        throw new WarpScriptException(getName() + " has been implemented with an unknown comparator.");
+    }
+  }
+
+  @Override
+  public List<GeoTimeSerie> filter(Map<String,String> labels, List<GeoTimeSerie>... series) throws WarpScriptException {
+    List<GeoTimeSerie> retained = new ArrayList<GeoTimeSerie>();
+
+    for (List<GeoTimeSerie> gtsinstances: series) {
+      for (GeoTimeSerie serie: gtsinstances) {
+        boolean found = false;
+        int i = 0;
+
+        while(!found && i < serie.size()) {
+          Object val = GTSHelper.valueAtIndex(serie, i++);
+
+          switch (type) {
+            case LONG:
+              found = verify(((Long) threshold).compareTo(((Number) val).longValue()));
+              break;
+            case DOUBLE:
+              found = verify(((Double) threshold).compareTo(((Number) val).doubleValue()));
+              break;
+            case STRING:
+              found = verify(((String) threshold).compareTo(val.toString()));
+              break;
+            case BOOLEAN:
+              found = ((Boolean) threshold).equals(val) ^ this.comparator == Comparator.NE;
+              break;
+          }
+        }
+
+        // In case of wanting the complementSet, the logic is reversed.
```

Comment:
Be more expressive in the description, the loop above exits when one datapoint which satisfies the comparison is found, in the case of an 'all' match, the operator checked is the opposite of the wanted one, therefore when one datapoint matches the opposite comparison, it means that not all the datapoints satisfy it and therefore, as the comparison operator is the opposite of the one we really want, it means that not all the datapoints satisfy the opposite comparison either and therefore we do not retain the series. The boolean logic is not enough of an explanation for the person discovering the code.